src/printer.rs

```diff
@@ -46,6 +46,23 @@ impl PrinterModel {
             PrinterModel::QL820NWB => 0x209d,
         }
     }
+
+    pub(crate) fn from_product_id(product_id: u16) -> Option<Self> {
```

Comment:
We now have two lookups, one from enum to pid, one from pid to enum... not sure if there is a nicer way to implement this, the duplication of the pids feels kind of wrong.